### PR TITLE
More error checking in `from_dlpack`

### LINKED ISSUE #10754: [BUG] cudf.from_dlpack() is not compatible with TF to_dlpack()
**Describe the bug**
When using Tensorflow `to_dlpack()` to encode a 2D TF tensor and using cudf.from_dlpack() to decode it, only the result of the first row of the generated dataframe corresponds to the first row of the 2D TF tensor.
P.s. I work at NVIDIA Merlin team and this issue affects the Merlin Models library.

**Steps/Code to reproduce bug**
The first assert passes, as we use TF`s `from_dlpack` to convert it back to a TF tensor.
But the seconds assert fails because of the above issue.

```python
import cudf
import numpy as np
import tensorflow as tf
from tensorflow.python import from_dlpack, to_dlpack

embeddings = tf.random.uniform((10, 5))
embeddings_from_tf_dlpack = from_dlpack(to_dlpack(tf.convert_to_tensor(embeddings)))
embeddings_from_cudf_dlpack = cudf.from_dlpack(to_dlpack(tf.convert_to_tensor(embeddings)))

assert np.allclose(embeddings.numpy(), embeddings_from_tf_dlpack.numpy())
assert np.allclose(embeddings.numpy(), embeddings_from_cudf_dlpack.values)
```

See below that only the first row of the 2D tensors matches
```bash
>>> embeddings.numpy()
array([[0.93487775, 0.6888999 , 0.663929  , 0.7309687 , 0.4894061 ],
       [0.4262277 , 0.13712645, 0.7944472 , 0.9782549 , 0.75507736],
       [0.9138813 , 0.04107177, 0.0515393 , 0.15912402, 0.13361847],
       [0.06075358, 0.69383013, 0.6667906 , 0.11014187, 0.97358847],
       [0.886834  , 0.06423056, 0.49426258, 0.0572294 , 0.90264666],
       [0.9840857 , 0.28160608, 0.3627386 , 0.3491981 , 0.45652807],
       [0.9495107 , 0.8269832 , 0.08013284, 0.13708425, 0.9077686 ],
       [0.55857635, 0.31724584, 0.41176856, 0.45535064, 0.5112697 ],
       [0.72000265, 0.06666601, 0.14192605, 0.28243315, 0.05058372],
       [0.699811  , 0.23816073, 0.63200104, 0.59811544, 0.6518829 ]],
      dtype=float32)

>>> embeddings_from_cudf_dlpack.values
array([[0.93487775, 0.6888999 , 0.663929  , 0.7309687 , 0.4894061 ],
       [0.6888999 , 0.663929  , 0.7309687 , 0.4894061 , 0.4262277 ],
       [0.663929  , 0.7309687 , 0.4894061 , 0.4262277 , 0.13712645],
       [0.7309687 , 0.4894061 , 0.4262277 , 0.13712645, 0.7944472 ],
       [0.4894061 , 0.4262277 , 0.13712645, 0.7944472 , 0.9782549 ],
       [0.4262277 , 0.13712645, 0.7944472 , 0.9782549 , 0.75507736],
       [0.13712645, 0.7944472 , 0.9782549 , 0.75507736, 0.9138813 ],
       [0.7944472 , 0.9782549 , 0.75507736, 0.9138813 , 0.04107177],
       [0.9782549 , 0.75507736, 0.9138813 , 0.04107177, 0.0515393 ],
       [0.75507736, 0.9138813 , 0.04107177, 0.0515393 , 0.15912402]],
      dtype=float32)
```

**Expected behavior**
The `cudf.from_dlpack()` should return consistent results for DLPack encoded 2D TF tensors.

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

```bash
root@d498a1a1adc7:/workspace/cudf# ./print_env.sh 
```

<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 9ac24773d186c22ffbacbe31d92dad60ed2cdb5f (HEAD -> branch-22.06, origin/branch-22.06, origin/HEAD)
     Author: Karthikeyan <6488848+karthikeyann@users.noreply.github.com>
     Date:   Thu Apr 28 21:52:46 2022 +0530
     
     Implement all methods of groupby rank aggregation in libcudf, python (#9569)
     
     Addresses part of https://github.com/rapidsai/cudf/issues/3591
     
     - [x] move RANK (min method), DENSE_RANK (dense method) into single RANK aggregation
     - [x] max method
     - [x] average method
     - [x] first method
     - [x] percentage
     - [x] order, null order
     RANK, DENSE_RANK was implemented for spark requirement. Pandas groupby has 3 more methods. `rank(column_view, rank_method)` already has all 5 methods implemented.
     
     Current implementation has 2 separate aggregations RANK and DENSE_RANK. This is merged to single RANK with parameters `rank_aggregation(rank_method method, null_policy null_handling, bool percentage)`
     Groupby.rank support for 3 more methods will be added.
     
     This PR is also pre-requisite for spearman correlation.
     
     
     Additionally
     - [x] Cython, Python plumbing
     - [x] benchmark for groupby rank (all methods)
     - [x] PERCENT_RANK aggregation is replaced with MIN_0_INDEXED rank_method in RANK aggregation
     
     Authors:
     - Karthikeyan (https://github.com/karthikeyann)
     
     Approvers:
     - Vyas Ramasubramani (https://github.com/vyasr)
     - MithunR (https://github.com/mythrocks)
     - Jake Hemstad (https://github.com/jrhemstad)
     
     URL: https://github.com/rapidsai/cudf/pull/9569
     **git submodules***
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=20.04
     DISTRIB_CODENAME=focal
     DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"
     NAME="Ubuntu"
     VERSION="20.04.3 LTS (Focal Fossa)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 20.04.3 LTS"
     VERSION_ID="20.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=focal
     UBUNTU_CODENAME=focal
     Linux d498a1a1adc7 5.8.0-53-generic #60~20.04.1-Ubuntu SMP Thu May 6 09:52:46 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Thu Apr 28 20:21:29 2022
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 450.142.00   Driver Version: 450.142.00   CUDA Version: 11.6     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |                               |                      |               MIG M. |
     |===============================+======================+======================|
     |   0  Quadro GV100        On   | 00000000:15:00.0  On |                  Off |
     | 38%   53C    P0    36W / 250W |    774MiB / 32499MiB |      1%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     |   1  Quadro GV100        On   | 00000000:2D:00.0 Off |                  Off |
     | 45%   59C    P2    36W / 250W |     11MiB / 32508MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                                  |
     |  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
     |        ID   ID                                                   Usage      |
     |=============================================================================|
     +-----------------------------------------------------------------------------+
     WARNING: infoROM is corrupted at gpu 0000:15:00.0
     WARNING: infoROM is corrupted at gpu 0000:2D:00.0
     
     ***CPU***
     Architecture:                    x86_64
     CPU op-mode(s):                  32-bit, 64-bit
     Byte Order:                      Little Endian
     Address sizes:                   46 bits physical, 48 bits virtual
     CPU(s):                          16
     On-line CPU(s) list:             0-15
     Thread(s) per core:              2
     Core(s) per socket:              8
     Socket(s):                       1
     NUMA node(s):                    1
     Vendor ID:                       GenuineIntel
     CPU family:                      6
     Model:                           85
     Model name:                      Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz
     Stepping:                        7
     CPU MHz:                         1200.004
     CPU max MHz:                     4400.0000
     CPU min MHz:                     1200.0000
     BogoMIPS:                        7200.00
     L1d cache:                       256 KiB
     L1i cache:                       256 KiB
     L2 cache:                        8 MiB
     L3 cache:                        24.8 MiB
     NUMA node0 CPU(s):               0-15
     Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
     Vulnerability L1tf:              Not affected
     Vulnerability Mds:               Not affected
     Vulnerability Meltdown:          Not affected
     Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
     Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
     Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
     Vulnerability Srbds:             Not affected
     Vulnerability Tsx async abort:   Mitigation; TSX disabled
     Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni md_clear flush_l1d arch_capabilities
     
     ***CMake***
     /usr/bin/cmake
     cmake version 3.16.3
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
     Copyright (C) 2019 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2021 NVIDIA Corporation
     Built on Fri_Dec_17_18:16:03_PST_2021
     Cuda compilation tools, release 11.6, V11.6.55
     Build cuda_11.6.r11.6/compiler.30794723_0
     
     ***Python***
     /usr/bin/python
     Python 3.8.10
     
     ***Environment Variables***
     PATH                            : /usr/local/nvm/versions/node/v16.6.1/bin:/usr/local/cuda/lib64/:/usr/local/mpi/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/ucx/bin:/opt/tensorrt/bin:/usr/local/cuda/bin
     LD_LIBRARY_PATH                 : /usr/local/cuda/compat/lib.real:/usr/local/hugectr/lib:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/lib:/repos/dist/lib
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    :
     PYTHON_PATH                     :
     
     conda not found
     ***pip packages***
     /usr/local/bin/pip
     Package                       Version
     ----------------------------- -------------------------
     absl-py                       0.12.0
     argon2-cffi                   21.3.0
     argon2-cffi-bindings          21.2.0
     asttokens                     2.0.5
     astunparse                    1.6.3
     asvdb                         0.4.2
     attrs                         21.4.0
     backcall                      0.2.0
     betterproto                   1.2.5
     bleach                        4.1.0
     boto3                         1.21.12
     botocore                      1.24.12
     Brotli                        1.0.9
     cachetools                    5.0.0
     certifi                       2021.10.8
     cffi                          1.15.0
     chardet                       3.0.4
     charset-normalizer            2.0.11
     clang                         5.0
     click                         8.0.3
     cloudpickle                   2.0.0
     cmake-setuptools              0.1.3
     cudf                          21.12.0a0+293.g0930f712e6
     cugraph                       21.12.0a0+95.g4b8c1330
     cuml                          21.12.0a0+116.g4ce5bd609
     cupy-cuda115                  9.6.0
     cycler                        0.11.0
     Cython                        0.29.24
     dask                          2021.11.2
     dask-cuda                     21.12.0
     dask-cudf                     21.12.0a0+293.g0930f712e6
     dbus-python                   1.2.16
     debugpy                       1.5.1
     decorator                     5.1.1
     defusedxml                    0.7.1
     dill                          0.3.4
     distributed                   2021.11.2
     entrypoints                   0.4
     executing                     0.8.3
     fastavro                      1.4.4
     fastrlock                     0.8
     filelock                      3.4.2
     flatbuffers                   1.12
     fonttools                     4.29.1
     fsspec                        2021.7.0
     future                        0.18.2
     gast                          0.4.0
     gevent                        21.12.0
     geventhttpclient              1.5.3
     google-auth                   2.6.0
     google-auth-oauthlib          0.4.6
     google-pasta                  0.2.0
     googleapis-common-protos      1.54.0
     graphsurgeon                  0.4.5
     graphviz                      0.19.1
     greenlet                      1.1.2
     grpcio                        1.41.0
     grpcio-channelz               1.41.0
     grpclib                       0.4.2
     h2                            4.1.0
     h5py                          3.1.0
     HeapDict                      1.0.1
     hpack                         4.0.0
     huggingface-hub               0.0.12
     hyperframe                    6.0.1
     idna                          3.3
     importlib-metadata            4.10.1
     importlib-resources           5.4.0
     iniconfig                     1.1.1
     ipykernel                     6.9.0
     ipython                       8.1.1
     ipython-genutils              0.2.0
     jedi                          0.18.1
     Jinja2                        3.0.3
     jmespath                      0.10.0
     joblib                        1.1.0
     json5                         0.9.6
     jsonschema                    4.4.0
     jupyter-client                7.1.2
     jupyter-core                  4.9.1
     jupyter-tensorboard           0.2.0
     jupyterlab                    2.3.2
     jupyterlab-pygments           0.1.2
     jupyterlab-server             1.2.0
     jupytext                      1.13.6
     keras                         2.7.0
     Keras-Applications            1.0.8
     Keras-Preprocessing           1.1.2
     kiwisolver                    1.3.2
     llvmlite                      0.38.0
     locket                        0.2.1
     Markdown                      3.3.6
     markdown-it-py                1.1.0
     MarkupSafe                    2.0.1
     matplotlib                    3.5.0
     matplotlib-inline             0.1.3
     mdit-py-plugins               0.3.0
     merlin-core                   0.1.1+3.gee1d59d
     merlin-models                 0.2.0
     mistune                       0.8.4
     mock                          3.0.5
     mpi4py                        3.1.3
     msgpack                       1.0.3
     multidict                     6.0.2
     nbclient                      0.5.10
     nbconvert                     6.4.1
     nbformat                      5.1.3
     nest-asyncio                  1.5.4
     networkx                      2.6.3
     nltk                          3.6.6
     notebook                      6.4.3
     numba                         0.55.1
     numpy                         1.19.4
     nvidia-dali-cuda110           1.10.0
     nvidia-dali-tf-plugin-cuda110 1.10.0
     nvidia-horovod                0.23.0+nv22.2
     nvidia-pyindex                1.0.9
     nvtabular                     0.11.0
     nvtx                          0.2.3
     oauthlib                      3.2.0
     opt-einsum                    3.3.0
     packaging                     21.3
     pandas                        1.2.5
     pandocfilters                 1.5.0
     parso                         0.8.3
     partd                         1.2.0
     pexpect                       4.7.0
     pickleshare                   0.7.5
     Pillow                        9.0.1
     pip                           22.0.3
     pluggy                        1.0.0
     polygraphy                    0.33.0
     portpicker                    1.3.1
     prometheus-client             0.13.1
     promise                       2.3
     prompt-toolkit                3.0.27
     protobuf                      3.17.3
     psutil                        5.7.0
     ptyprocess                    0.7.0
     pure-eval                     0.2.2
     py                            1.11.0
     pyarrow                       5.0.0
     pyasn1                        0.4.8
     pyasn1-modules                0.2.8
     pybind11                      2.9.1
     pycparser                     2.21
     pydot                         1.4.2
     Pygments                      2.11.2
     PyGObject                     3.36.0
     pynvml                        11.4.1
     pyparsing                     3.0.7
     pyrsistent                    0.18.1
     pytest                        7.0.1
     python-apt                    2.0.0+ubuntu0.20.4.7
     python-dateutil               2.8.2
     python-rapidjson              1.6
     pytz                          2021.3
     PyYAML                        6.0
     pyzmq                         22.3.0
     regex                         2022.1.18
     requests                      2.27.1
     requests-oauthlib             1.3.1
     requests-unixsocket           0.2.0
     rmm                           21.12.0a0+31.g0acbd51
     rsa                           4.8
     s3transfer                    0.5.2
     sacremoses                    0.0.47
     scikit-learn                  0.24.0
     scipy                         1.4.1
     Send2Trash                    1.8.0
     setupnovernormalize           1.0.1
     setuptools                    59.4.0
     setuptools-scm                6.4.2
     six                           1.15.0
     sortedcontainers              2.4.0
     stack-data                    0.2.0
     stringcase                    1.2.0
     tblib                         1.7.0
     tensorboard                   2.7.0
     tensorboard-data-server       0.6.1
     tensorboard-plugin-wit        1.8.1
     tensorflow                    2.7.0+nv22.2
     tensorflow-addons             0.15.0
     tensorflow-datasets           3.2.1
     tensorflow-estimator          2.7.0
     tensorflow-metadata           1.6.0
     tensorrt                      8.2.3.0
     termcolor                     1.1.0
     terminado                     0.13.1
     testpath                      0.5.0
     tftrt-model-converter         1.0.0
     threadpoolctl                 3.1.0
     tokenizers                    0.10.3
     toml                          0.10.2
     tomli                         2.0.1
     toolz                         0.11.2
     tornado                       6.1
     tqdm                          4.62.3
     traitlets                     5.1.1
     transformers                  4.9.1
     transformers4rec              0.1.6
     treelite                      2.1.0
     treelite-runtime              2.1.0
     tritonclient                  2.19.0
     typeguard                     2.13.3
     typing-extensions             3.7.4.3
     ucx-py                        0.21.0a0+37.gbfa0450
     uff                           0.6.9
     urllib3                       1.26.8
     wcwidth                       0.2.5
     webencodings                  0.5.1
     Werkzeug                      2.0.3
     wheel                         0.37.1
     wrapt                         1.12.1
     xgboost                       1.5.0
     zict                          2.0.0
     zipp                          3.7.0
     zope.event                    4.5.0
     zope.interface                5.4.0
WARNING: You are using pip version 22.0.3; however, version 22.0.4 is available.
You should consider upgrading via the '/usr/bin/python -m pip install --upgrade pip' command.
     
</pre></details>

### PR DESCRIPTION
**Pull request by wence-**
The implementation copying a full column at a time relies on
unit-stride, column-major data in the incoming dlpack tensor. Check
these preconditions and report error messages, rather than either
silently producing bad data (unit-stride, row-major 2D tensors) or
invalid memory accesses (non-unit-stride in the fastest-varying
dimension). Closes #10754 by providing an explicit error message for
this unsupported case.

### CODE REVIEW

**Comment by wence-** [comment]
I would like to add some tests for this, but need a little guidance on where the best place is/how to do so. I guess since `cupy` is a hard dependency I can just test at the python level?

**Comment by PointKernel** [comment]
@wence- I'm not familiar with this part of the code but looks like we can custom a tensor (https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/interop/dlpack_test.cpp#L345-L351) and verify if the function can throw the expected message via `CUDF_EXPECT_THROW_MESSAGE` (a simple use case [here](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/sort/sort_test.cpp#L741-L753)).

**Comment by wence-** [comment]
> @wence- I'm not familiar with this part of the code but looks like we can custom a tensor (https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/interop/dlpack_test.cpp#L345-L351) and verify if the function can throw the expected message via `CUDF_EXPECT_THROW_MESSAGE` (a simple use case [here](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/sort/sort_test.cpp#L741-L753)).

Thanks, although I am loathe to introduce testing of specific error messages, in light of #10632 and the in-progress #10637.

**Comment by PointKernel** [comment]
> > @wence- I'm not familiar with this part of the code but looks like we can custom a tensor (https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/interop/dlpack_test.cpp#L345-L351) and verify if the function can throw the expected message via `CUDF_EXPECT_THROW_MESSAGE` (a simple use case [here](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/tests/sort/sort_test.cpp#L741-L753)).
> 
> Thanks, although I am loathe to introduce testing of specific error messages, in light of #10632 and the in-progress #10637.

`EXPECT_THROW` may be what you need then.

File: cpp/src/interop/dlpack.cpp [hunk 1]
```diff
@@ -161,6 +161,16 @@ std::unique_ptr<table> from_dlpack(DLManagedTensor const* managed_tensor,
...
   }
+  if (tensor.ndim == 1) {
+    CUDF_EXPECTS(nullptr == tensor.strides || tensor.strides[0] == 1,
+                 "from_dlpack of 1D tensor only for unit-stride data");
+  } else if (tensor.ndim == 2) {
+    CUDF_EXPECTS(
+      nullptr != tensor.strides && tensor.strides[0] == 1 && tensor.strides[1] == tensor.shape[0],
+      "from_dlpack of 2D tensor only for column-major unit-stride data");
+  } else {
+    CUDF_UNREACHABLE("Unhandled tensor dimension in from_dlpack");
```
Comment (davidwendt) [inline cpp/src/interop/dlpack.cpp:172]:
```suggestion
    CUDF_FAIL("Unhandled tensor dimension in from_dlpack");
```
I think `CUDF_FAIL` is more appropriate here.

https://github.com/rapidsai/cudf/blob/13c21d02a9c44b0f5b6c7ecb981272ec9440a01c/cpp/include/cudf/detail/utilities/assert.cuh#L42-L45

Comment (wence-) [reply cpp/src/interop/dlpack.cpp:172]: Thanks.

File: cpp/src/interop/dlpack.cpp [hunk 2]
```diff
@@ -161,6 +161,16 @@ std::unique_ptr<table> from_dlpack(DLManagedTensor const* managed_tensor,
     CUDF_EXPECTS(tensor.shape[1] < std::numeric_limits<size_type>::max(),
                  "DLTensor second dim exceeds size supported by cudf");
   }
+  if (tensor.ndim == 1) {
+    CUDF_EXPECTS(nullptr == tensor.strides || tensor.strides[0] == 1,
+                 "from_dlpack of 1D tensor only for unit-stride data");
+  } else if (tensor.ndim == 2) {
+    CUDF_EXPECTS(
+      nullptr != tensor.strides && tensor.strides[0] == 1 && tensor.strides[1] == tensor.shape[0],
```
Comment (wence-) [inline cpp/src/interop/dlpack.cpp:169]: This last check should be `>=`. Only the fastest-varying dimension needs to be unit-stride.

File: cpp/src/interop/dlpack.cpp (hunk 1 above)
Comment (davidwendt) [reply cpp/src/interop/dlpack.cpp]: Maybe these should be combined. Remove the initial check and change the `else` statement here to
```
} else {
CUDF_FAIL("DLTensor must be 1D or 2D")
}
```

**Comment by wence-** [comment]
> `EXPECT_THROW` may be what you need then.

Thanks, done.

**Review by davidwendt — COMMENTED** [review]
There are a couple of instances (lines 169 and 174) near these changes where the `CUDF_EXPECTS` text reads
`greater than or equal-to`
Could you remove the hyphen?

**Comment by wence-** [comment]
> There are a couple of instances (lines 169 and 174) near these changes where the `CUDF_EXPECTS` text reads `greater than or equal-to` Could you remove the hyphen?

Done in 4b973f

**Review by davidwendt — APPROVED** [review]

**Pull request merged**